src/server/editorServices.ts

```diff
@@ -541,174 +576,162 @@ namespace ts.server {
         }
 
         private updateProjectGraphs(projects: Project[]) {
-            let shouldRefreshInferredProjects = false;
             for (const p of projects) {
                 if (!p.updateGraph()) {
-                    shouldRefreshInferredProjects = true;
+                    this.pendingInferredProjectUpdate = true;
                 }
             }
-            if (shouldRefreshInferredProjects) {
-                this.refreshInferredProjects();
-            }
         }
 
-        private onSourceFileChanged(fileName: NormalizedPath) {
+        private onSourceFileChanged(fileName: NormalizedPath, eventKind: FileWatcherEventKind) {

```

Comment:
We only test `eventKind` against one of the 3 event kinds. Would be more straightforward for this to take a boolean `isDelete`.